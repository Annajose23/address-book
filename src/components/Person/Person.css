.person {
    width: 500px;
    padding: 16px;
    text-align: center;
    border: 1px solid #eee;
    display: inline-block;
    margin: 10px;
    box-sizing: border-box;
    cursor: pointer;
    border-radius: 5px;
}

.person:hover,
.person:active {
    background-color: #242526;
}

