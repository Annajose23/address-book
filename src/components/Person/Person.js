import React from "react";
import './Person.css';

const person = (props) => {
  return (
    <article className="person" onClick={props.clicked}>
        {props.name.title}. {props.name.first} {props.name.last}
    </article>
  );
};

export default person;
