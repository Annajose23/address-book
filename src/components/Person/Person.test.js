import React from 'react';
import renderer from 'react-test-renderer'; 
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import Person from './Person';

configure({ adapter: new Adapter() });

describe('person', () => {
    let wrapper;
    const props = {
        name:{
            title:'Ms',
            first:'anna',
            last:'jose'
        },
        cell:'909876543'
    }
    beforeEach(() => {
        wrapper = shallow(<Person name={props.name} clicked={()=>{}}/>);
    });

    it('should render person component with props', ()=>{
        expect(wrapper).not.toBeNull();
    });

    it('should contain article tag', ()=>{
        expect(wrapper.find('article')).toHaveLength(1);
    });

    it('render correctly Person component', () => {  
        const PersonComponent = renderer.create(<Person name={props.name} clicked={()=>{}}/>).toJSON();
        expect(PersonComponent).toMatchSnapshot();
    });

    it("should contain article tag with classname person", () => {
        expect(wrapper.find("article").hasClass("person")).toEqual(true);
      });
})