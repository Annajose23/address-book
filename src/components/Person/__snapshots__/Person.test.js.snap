// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`person render correctly Person component 1`] = `
<article
  className="person"
  onClick={[Function]}
>
  Ms
  . 
  anna
   
  jose
</article>
`;
